content/docs/getting-started/hello-world.md

```diff
@@ -71,93 +56,74 @@ Next, we'll add some to the `client` `TcpStream`. This asynchronous task now cre
 the stream and then yields it once it's been created for additional processing.
 
 ```rust
-# #![deny(deprecated)]
-# extern crate tokio;
-#
-# use tokio::net::TcpStream;
-# use tokio::prelude::*;
-# fn main() {
-# let addr = "127.0.0.1:6142".parse().unwrap();
-let client = TcpStream::connect(&addr).and_then(|stream| {
-    println!("created stream");
-
-    // Process stream here.
-
-    Ok(())
-})
-.map_err(|err| {
-    // All tasks must have an `Error` type of `()`. This forces error
-    // handling and helps avoid silencing failures.
-    //
-    // In our example, we are only going to log the error to STDOUT.
-    println!("connection error = {:?}", err);
-});
-# }
+use tokio::net::TcpStream;
+
+fn main() {
+    let addr = "127.0.0.1:6142";
+    let client = TcpStream::connect(addr);
+
+    let future = async move {
+        match client.await {
+            Ok(_stream) => {
+                println!("created stream");
+
+                // Process stream here.
+            }
+            Err(err) => {
+                // In our example, we are only going to log the error to STDOUT.
+                println!("connection error = {:?}", err);
+            }
+        };
+    };
+}
```

Comment:
In this example, the future is never run. I think that, in the previous example, that is okay, since we hid the `fn main()` and it just looked like code somewhere inside a program. Here, though, we are showing `fn main()` at the top of the example. IMO, this gives the incorrect implication that users can write code like this and expect it to work.

Maybe we should hide the `fn main()` in this example?